src/plugins/analysis/linter/code/source_code_analysis.py

```diff
@@ -37,37 +40,37 @@ class AnalysisPlugin(AnalysisBasePlugin):
 
     def __init__(self, plugin_adminstrator, config=None, recursive=True, offline_testing=False):
         self.config = config
+        if not self._check_docker_installed():
+            raise RuntimeError('Docker is not installed.')
         super().__init__(plugin_adminstrator, config=config, plugin_path=__file__, recursive=recursive, offline_testing=offline_testing)
 
-    def _determine_script_type(self, file_object: FileObject):
-        '''
-        Indicators:
-        1. file_type full includes shell, python etc.
-        2. shebang #!/bin/sh, #!/usr/bin/env python etc.
-        3. file ending *.sh, *.py etc.
-        '''
-        full_file_type = file_object.processed_analysis['file_type']['full'].lower()
-        for script_type in self.SCRIPT_TYPES:
-
-            if self.SCRIPT_TYPES[script_type]['mime'] in full_file_type.lower():
-                return script_type
+    @staticmethod
+    def _check_docker_installed():
+        _, return_code = execute_shell_command_get_return_code('docker -v')
+        return return_code == 0
 
-            first_line = file_object.binary.decode().splitlines(keepends=False)[0]
-            if first_line.find(self.SCRIPT_TYPES[script_type]['shebang']) >= 0:
+    def _get_script_type(self, file_object, linguist_output):
+        if 'language' in linguist_output:
+            script_language = linguist_output.split('language:', 1)[1]
+            script_type = script_language.translate({ord(c): None for c in string.whitespace}).lower()
+            if script_type:
+                file_object.processed_analysis['file_type']['linguist'] = script_type
                 return script_type
```

Comment:
This will not work correctly if there is additional linguist output after the "language" line (e.g. "appears to be generated source code")
In my opinion it would be better to store the full linguist output in the file_type analysis (because there could be more interesting information in it for the user).
Also this seems unnecessary complicated and could be easier parsed with re.search, like e.g.
```suggestion
            file_object.processed_analysis['file_type']['linguist'] = linguist_output
            match = re.search(r'language:\s*(\w+)', linguist_output)
            if match:
                return match.groups()[0].lower()
```